src/bank.rs

```diff
@@ -212,6 +216,25 @@ impl Bank {
         }
     }
 
+    /// Return the position of the last_id in the last_id_queue starting from the back
+    /// If the last_id is not found last_id_queue.len() is returned
+    fn compute_entry_id_age(last_id_queue: &VecDeque<Hash>, entry_id: Hash) -> Option<usize> {
+        for (i, id) in last_id_queue.iter().rev().enumerate() {
```

Comment:
is this as efficient as I'd hope?

Reply:
what if last_ids was a hashmap (id -> nth_inserted) and the bank had a monotonically increasing nth_last_id counter?

entry_id_age would then be constant time, and I could chain lookups in the forking bank

we could combine last_id_sigs and last_ids_queue?

Reply:
You think that will be faster then an array of a few hundred items?  I guess the ids are pretty big so iteration might not be great

Reply:
for the forking bank, a hashmap would be easier to deal with...

leave as-is and I'll work on top of this, prolly change to a hashmap, have to figure out when best to call retain()...